apitools/base/py/list_pager.py

```diff
@@ -52,14 +52,15 @@ def YieldFromList(
           response message holding the page token for the next page.
       batch_size_attribute: str, The name of the attribute in a
           response message holding the maximum number of results to be
-          returned.
+          returned. None if caller-specified batch size is unsupported.
 
     Yields:
       protorpc.message.Message, The resources listed by the service.
 
     """
     request = encoding.CopyProtoMessage(request)
-    setattr(request, batch_size_attribute, batch_size)
+    if batch_size_attribute:
```

Comment:
Done. Note to avoid breaking existing code, unset still defaults to 'maxResults' as the field name. Callers must explicitly set it to None to indicate no field should be set (presumably because no such field exists).